### PR TITLE
refactor: Form fields now correctly implement FormField

### PR DESCRIPTION
**Pull request by mikecoomber**

### CODE REVIEW

File: lib/src/components/search_bar/search_bar.dart [hunk 1]
```diff
@@ -1,64 +1,198 @@
...
+class ZetaSearchBar extends ZetaTextFormField {
   /// Constructor for [ZetaSearchBar].
-  const ZetaSearchBar({
-    super.key,
-    this.size,
-    this.shape,
-    this.hint,
-    this.initialValue,
-    this.onChanged,
-    this.onSubmit,
```
Comment (thelukewalton) [inline lib/src/components/search_bar/search_bar.dart:16]: I think onSumbit need to be deprecated

File: lib/src/components/text_input/internal_text_input.dart [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+  }
+}
+
+/// The current state of a [InternalTextInput]
+class InternalTextInputState extends State<InternalTextInput> {
+  late final TextEditingController _controller;
+  ZetaColors get _colors => Zeta.of(context).colors;
+
+  // TODO(mikecoomber): refactor to use WidgetStateController
+  bool _hovered = false;
```
Comment (thelukewalton) [inline lib/src/components/text_input/internal_text_input.dart:169]: Should we make a jira ticket for this?

File: test/src/components/search_bar/golden/search_bar_default.png
[the remark this answers is not in the document]
Comment (mikecoomber) [reply test/src/components/search_bar/golden/search_bar_default.png:1]: I fixed the styling on the placeholder, could that have done it?

[the remark this answers is not in the document]
Comment (thelukewalton) [reply test/src/components/search_bar/golden/search_bar_default.png:1]: I guess it must be

**Pull request merged**